domain/application/state/application.go

```diff
@@ -683,21 +824,75 @@ DELETE FROM net_node WHERE uuid IN
 	return nil
 }
 
-func (st *ApplicationState) deleteCloudContainer(ctx context.Context, tx *sqlair.TX, unitID string) error {
-	unit := coreUnit{ID: unitID}
+func (st *ApplicationState) deleteCloudContainer(ctx context.Context, tx *sqlair.TX, netNodeID string) error {
+	cloudContainer := cloudContainer{NetNodeID: netNodeID}
+
+	if err := st.deleteCloudContainerPorts(ctx, tx, netNodeID); err != nil {
+		return errors.Trace(err)
+	}
+
+	if err := st.deleteCloudContainerAddresses(ctx, tx, netNodeID); err != nil {
+		return errors.Trace(err)
+	}
 
-	deleteCloudContainer := `DELETE FROM cloud_container WHERE net_node_uuid = $coreUnit.uuid`
-	deleteCloudContainerStmt, err := st.Prepare(deleteCloudContainer, unit)
+	deleteCloudContainerStmt, err := st.Prepare(`
+DELETE FROM cloud_container
+WHERE net_node_uuid = $cloudContainer.net_node_uuid`, cloudContainer)
 	if err != nil {
 		return errors.Trace(err)
 	}
 
-	if err := tx.Query(ctx, deleteCloudContainerStmt, unit).Run(); err != nil {
+	if err := tx.Query(ctx, deleteCloudContainerStmt, cloudContainer).Run(); err != nil {
 		return errors.Trace(err)
 	}
 	return nil
 }
 
+func (st *ApplicationState) deleteCloudContainerAddresses(ctx context.Context, tx *sqlair.TX, netNodeID string) error {
+	unit := coreUnit{
+		NetNodeID: netNodeID,
+	}
+	deleteAddressStmt, err := st.Prepare(`
+DELETE FROM ip_address
+WHERE device_uuid IN
+(SELECT device_uuid FROM link_layer_device lld
```

Comment:
The form I have seen us following for this is to leave the select on the same line as the in and indent everything up to the select.